core/core_modules/kpm.js

```diff
@@ -166,48 +170,42 @@ var git = require.once('../git.js'),
 
     update = function (module, api, event) {
         api.sendMessage('Updating "' + module.name + '" (' + module.version + ')...', event.thread_id);
-        git.pullWithPath(module.folderPath, function (err, consoleOutput) {
+        git.pullWithPath(module.folderPath, function (err) {
             if (err) {
                 api.sendMessage('Update failed. Manual intervention is probably required.', event.thread_id);
             } else {
-                api.sendMessage('Restarting module "' + module.name + '"...');
+                api.sendMessage('Restarting module "' + module.name + '"...', event.thread_id);
                 // unload the current version
-                cfg.saveModuleConfig(module.name);
                 this.loadedModules = this.loadedModules.filter(function (value) {
-                    if (value.name === module.name) {
-                        if (value.unload) {
-                            value.unload();
-                        }
-                        return false;
+                    if (value.name !== module.name) {
+                        return true;
                     }
-                    return true;
+                    exports.platform.modulesLoader.unloadModule(value);
+                    return false;
                 });
                 delete moduleCache[module.name];
 
                 // load new module copy
-                var m = require.once(path.join(module.folderPath, 'kassy.json'));
-                m.folderPath = module.folderPath;
-                moduleCache[module.name] = m;
-                module = m;
-                this.loadedModules.push(modules.loadModule(module));
-
-                api.sendMessage('"' + module.name + '" is now at version ' + module.version + '.', event.thread_id);
+                var descriptor = exports.platform.modulesLoader.verifyModule(module.folderPath),
+                    m = exports.platform.modulesLoader.loadModule(descriptor);
+                if (m !== null) {
+                    moduleCache[descriptor.name] = descriptor;
+                    exports.platform.loadedModules.push(m);
+                    api.sendMessage('"' + module.name + '" is now at version ' + module.version + '.', event.thread_id);
+                } else {
+                    api.sendMessage('Loading updated "' + module.name + '" failed. Manual intervention will be required.', event.thread_id);
+                }
             }
         }.bind(this));
     },
 
     uninstall = function(module, api, event) {
         api.sendMessage('Unloading module "' + module.name + '"...', event.thread_id);
         // unload the current version
-        cfg.saveModuleConfig(module.name);
         this.loadedModules = this.loadedModules.filter(function (value) {
-            if (value.name === module.name) {
-                if (value.unload) {
-                    value.unload();
-                }
-                return false;
-            }
-            return true;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'return'.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962458/issues/source?bid=3282987&fileBranchId=3282998#l206)